src/modules/checkIsValidUserInput.js

```diff
@@ -8,8 +8,63 @@
  * @param {string} userInput - The user input
  * @return {boolean} - True if the user input is valid, false otherwise
  */
+const checkIsUniqueNumbers = (value) => {
+  const arrayValue = `${value}`.split('');
+  // console.log(arrayValue.length);
+
+  if (arrayValue.length !== 4) {
+    // console.log('(arrayValue.length !== 4)');
+
+    return false;
+  }
+
+  if (typeof +value !== 'number') {
+    // console.log(`(typeof value !== 'number')`);
+
+    return false;
+  }
+
+  if (arrayValue[0] === '0') {
+    // console.log(`(arrayValue[0] === '0')`);
+
+    return false;
+  }
+
+  const arrayValueToCheck = [...arrayValue];
+
+  for (let i = 0; i < arrayValue.length; i++) {
+    const numberToCheck = arrayValue[i];
+
+    arrayValueToCheck.splice(i, 1);
+
+    if (arrayValueToCheck.includes(numberToCheck)) {
+      // console.log(`arrayValueToCheck.includes(numberToCheck)`);
+
+      return false;
+    }
+
+    return true;
```

Comment:
The `return true;` statement is inside the loop, causing the function to return true after checking only the first digit. Move this statement outside the loop to ensure all digits are checked for uniqueness before returning true.